### ISSUE TITLE
Suppress `CA1707` for unit test methods

### ISSUE THREAD

**Issue by Flash0ver** [Flash0ver/F0.Analyzers#96]
_CA1707: Identifiers should not contain underscores_
(see https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
is also reported for unit test methods.

```CSharp
[Fact]
public async Task [|ComputeRefactoringsAsync_CSharp9_TargetTypedNewExpression|]()
{
```

However, the [Unit testing best practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#best-practices) suggest:
> ...
> ### Naming your tests
>
> The name of your test should consist of three parts:
> - The name of the method being tested.
> - The scenario under which it's being tested.
> - The expected behavior when the scenario is invoked.
>
> #### Why?
>
> - Naming standards are important because they explicitly express the intent of the test.
>
> ...
>
> ```CSharp
> [Fact]
> public void Add_SingleNumber_ReturnsSameNumber()
> {
>     var stringCalculator = new StringCalculator();
>
>     var actual = stringCalculator.Add("0");
>
>     Assert.Equal(0, actual);
> }
> ```
>
> ...

Add a [DiagnosticSuppressor](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.diagnosticsuppressor) which suppresses _Warning CA1707_ for unit tests of all `Test/` templates from the _.NET SDK_:
- [MSTest](https://github.com/microsoft/testfx)
- [NUnit](https://github.com/nunit/nunit)
- [xUnit](https://github.com/xunit/xunit)

For testing of _DiagnosticSuppressor_, see dotnet/roslyn-sdk#942
